server/index.js

```diff
@@ -102,24 +138,49 @@ app.put('/roulette/add', (req, res, next) => {
       SET "details" = $2
       RETURNING *
   `;
-  const params1 = [restaurantId, req.body];
+  const params1 = [req.body.id, req.body];
   db.query(sql1, params1)
     .then(result => {
       const sql2 = `
         INSERT INTO "roulette" ("restaurantId", "accountId")
           VALUES ($1, $2)
           ON CONFLICT ("restaurantId", "accountId") DO NOTHING
       `;
-      const params2 = [restaurantId, TEMP_USER_ID];
+      const params2 = [req.body.id, TEMP_USER_ID];
       db.query(sql2, params2)
         .then(res.status(201).json(result.rows[0]))
         .catch(err => next(err));
     })
     .catch(err => next(err));
 });
 
-app.put('/favorites/add', (req, res, next) => {
-  const { id: restaurantId } = req.body;
+app.delete('/roulette/:id', (req, res, next) => {
+  if (!req.params.id) throw new ClientError(400, 'id is a required field.');
```

Comment:
`req.params.id` is guaranteed to be truthy, otherwise the pattern `/roulette/:id` would not have been matched. I think it's safe to just remove this guard.